### ISSUE TITLE
[jvm-package] how to save predict result file to hdfs?

### ISSUE THREAD

**Issue by leoli2012** [dmlc/xgboost#1040]
hi ,i want save the result as a text file that predict by xgboost4j-spark, predict() method accept a RDD[Vector] and return a RDD[Array[Array[Float]] object , RDD have a saveAsTextFile api, i try this but nothing saved, i think element in RDD[Array[Array[Float]] is a array object, so saveAsTextFile will save the array object itself, not the float element in Array[Array[Float]].

**Comment by CodingCat** [comment]
saveAsTextFile essentially calls `toString` of the elements, in Scala, the result of toString of Array is something like `A@memoryAddr`

you can do a map before saveAsTextFile

**Comment by CodingCat** [comment]
sorry, closed it mistakenly......

**Comment by leoli2012** [comment]
@CodingCat ， so the predict value for each sample(a row in test data set)is not a float number, but a float array?

**Comment by CodingCat** [comment]
yes, this is to accommodate the situation that we have the predicted leaf indices from all trees....

**Comment by tqchen** [comment]
and also used in multi-class classification

**Comment by leoli2012** [comment]
@CodingCat , i see,  if i want the predict result ,do i must computed by myself depend on these predict array?

**Comment by leoli2012** [comment]
@CodingCat , i have no idea about the predict result generator, below is a part of my code
val train_data = MLUtils.loadLibSVMFile(sc, "hdfs://192.168.2.170:9000/train_model_off_line_test/output/model_src_true1/*")
 val sample_10 = MLUtils.loadLibSVMFile(sc,"hdfs://192.168.2.170:9000/train_model_off_line_test/output/sample_10")
val paramMap = List(
      "eta" -> 0.1f,
      "max_depth" -> 3,
      "objective" -> "binary:logistic").toMap
    val numwork = 9
    val gbdt_model = XGBoost.train(train_data, paramMap, numRound, nWorkers = numwork)
    val boster = gbdt_model.booster
    val model_dump = boster.getModelDump()
    for(i<- 0 to model_dump.length - 1)
        println(model_dump(i))    
    val eval_data = sample_10.map {point => point.features}           //sample_10 data is libsvm format, 
    //loadLibSVMFile return a RDD[LabledPoint] object, however, predict api just reccive RDD[Vector]
    val pred_result = gbdt_model.predict(eval_data)
    val result_file = pred_result.map{ point => point.map(_.toList).toList}
    result_file.saveAsTextFile("hdfs://192.168.2.170:9000/train_model_off_line_test/output/pred_result")

sample_10 have 10 row,namely ,10 sample data , and i get a below  result,  i was unable to understand what's that mean,
List(List(0.6856271), List(0.6856271), List(0.6856271), List(0.6856271), List(0.6856271))
List(List(0.6856271), List(0.6856271), List(0.6856271), List(0.6856271), List(0.6856271))

**Comment by CodingCat** [comment]
that means, your RDD contains 

* 2 partitions - so there are two outer List

* each partition contains 5 results - in each outer list, we have 5 inner List 

* each inner List shows the prediction result (for binary classification), that's 0.68+,

**Comment by leoli2012** [comment]
@CodingCat hi,thanks for you help, I'm confused about the result format, i use ""objective" -> "reg:logistic" and 6 sample to run, but i  get the same result format.
List(List(0.6272862), List(0.6272862), List(0.6272862), List(0.6272862))
List(List(0.6272862), List(0.6272862))
two list means two classification ? but i job is "reg:logistic", result should only 1 class

**Comment by CodingCat** [comment]
two list means your testset RDD contains two partitions, each maps to a DMatrix, and each DMatrix maps to a list 

each inner List represents a data sample, the 0.62x is the result

**Comment by leoli2012** [comment]
so ,there are two DMatrixs, first one have 4 predict result, and another have 2 result , the 4 predict in dmatrix map to 4 sample in testdata set?

**Comment by CodingCat** [comment]
yes

**Comment by smundhra** [comment]
How should one map these predicted results to the original data ?More specifically I would want to know the actual label for these predictions so that I can find the accuracy on my test set.
